### PR TITLE
BUILD.md: instructions for using clang

### PR DESCRIPTION
**Pull request by ajfabbri**
Took me a bit to figure this out, so I thought I'd add a hint to the BUILD.md.  Open to suggestions.

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -163,6 +163,17 @@ list some basic and more used commands as example.
     ./waf --targets tests/test_math
     ```
 
+* **Use clang instead of gcc**
```
Comment (andyp1per) [inline BUILD.md:166]: Clang is used automatically on MacOS - maybe you can make it clear that this is only for linux

Comment (ajfabbri) [reply BUILD.md:166]: Will do, thanks.

**Comment by ajfabbri** [comment]
Force push: mention defaults on linux versus MacOS.

**Review by andyp1per — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**